### PR TITLE
Adding watch later page

### PR DESCRIPTION
**Pull request by b0llu**
## What to check
- [ ] src/Context/WatchLater.context.js
- [ ] src/Pages/WatchLaterPage/WatchLaterPage.jsx

## What not to check
- [ ] Do Not Check Any Other Files

live Preview,
https://stormplay.netlify.app/

### CODE REVIEW

File: src/Context/WatchLater.context.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+const WatchLaterProvider = ({ children }) => {
+  const encodedToken = localStorage.getItem("StormPlayToken");
+  const { dispatch } = useReducerContext();
+
+  const addToWatchLater = async (video) => {
+    try {
+      const response = await axios.post(
+        "/api/user/watchlater",
+        { video },
+        { headers: { authorization: encodedToken } }
```
Comment (govindappaarun) [inline src/Context/WatchLater.context.js:17]: suggestion: add common API logic into a interceptor, and use constants file for the urls

File: src/Context/WatchLater.context.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      );
+      if (response.status === 201) {
+        SuccessToast("Added To Watch Later");
+        dispatch({ type: "WATCH_LATER", payload: response.data.watchlater });
+      }
+    } catch (error) {
+      console.log(error);
```
Comment (govindappaarun) [inline src/Context/WatchLater.context.js]: Suggestion: You can dispatch a failure toast

File: src/Hook/useFunction.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { useReducerContext } from "Context";
+
```
Comment (govindappaarun) [inline src/Hook/useFunction.js:2]: suggestion: think of a better name for this hook

File: src/Hook/useFunction.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    } else {
+      return videos.filter((v) =>
+        v.title.toLowerCase().includes(searchTerm.toLowerCase())
+      );
+    }
+  };
+
+  const searchFilteredVideos = searchFilter();
+
+  return { searchFilteredVideos };
```
Comment (govindappaarun) [inline src/Hook/useFunction.js:18]: Suggestion: naming to be improved

File: src/Pages/WatchLaterPage/WatchLaterPage.jsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import { Loader, VideoCard } from "Components";
+import { useReducerContext } from "Context";
+import "./WatchLaterPage.css";
+
+export const WatchLaterPage = () => {
+  const { watchLater, loading } = useReducerContext();
+
```
Comment (govindappaarun) [inline src/Pages/WatchLaterPage/WatchLaterPage.jsx:7]: suggestion: need better naming

**Review by govindappaarun — APPROVED** [review]
I see some improvements on naming and on reuse
otherwise things looks good to merge to me

File: src/Context/WatchLater.context.js (hunk 2 above)
Comment (b0llu) [reply src/Context/WatchLater.context.js]: thanks for pointing it out will make sure to dispatch a failure toast

File: src/Hook/useFunction.js (hunk 4 above)
Comment (b0llu) [reply src/Hook/useFunction.js:18]: will think on better naming

File: src/Pages/WatchLaterPage/WatchLaterPage.jsx (hunk 5 above)
Comment (b0llu) [reply src/Pages/WatchLaterPage/WatchLaterPage.jsx:7]: will think on better naming

File: src/Hook/useFunction.js (hunk 3 above)
Comment (b0llu) [reply src/Hook/useFunction.js:2]: will think on better naming

File: src/Context/WatchLater.context.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      value={{ addToWatchLater, removeFromWatchLater }}
+    >
+      {children}
+    </WatchLaterContext.Provider>
+  );
+};
+
+const useWatchLaterContext = () => useContext(WatchLaterContext);
+
+export { useWatchLaterContext, WatchLaterProvider };
```
Comment (gadgetfather) [inline src/Context/WatchLater.context.js:52]: Context is well written

**Review by gadgetfather — APPROVED** [review]
- Code is clean
- You can use toasts to show errors
Great Job @b0llu

File: src/Context/WatchLater.context.js (hunk 6 above)
Comment (b0llu) [reply src/Context/WatchLater.context.js:52]: thanks

**Pull request merged**